### ISSUE TITLE
ImportError: cannot import name 'Payment' from partially initialized module 

### ISSUE THREAD

**Issue by pennersr** [mollie/mollie-api-python#301]
When trying to import the `Payment` class you hit a circular import (tested with 3.0.0, 3.0.1, 3.1.0):

```
# python
Python 3.11.2 (main, Feb  9 2023, 00:38:19) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from mollie.api.objects.payment import Payment
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.11/site-packages/mollie/api/objects/payment.py", line 1, in <module>
    from ..resources import PaymentCaptures, PaymentChargebacks
  File "/usr/local/lib/python3.11/site-packages/mollie/api/resources/__init__.py", line 13, in <module>
    from .payments import *  # noqa: F401, F403
    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/mollie/api/resources/payments.py", line 6, in <module>
    from ..objects.payment import Payment
ImportError: cannot import name 'Payment' from partially initialized module 'mollie.api.objects.payment' (most likely due to a circular import) (/usr/local/lib/python3.11/site-packages/mollie/api/objects/payment.py)
```

Workaround:

```
>>> # Add this dummy import
>>> import mollie.api.resources.payments
>>> # Now this works:
>>> from mollie.api.objects.payment import Payment
```

**Comment by whyscream** [comment]
This is probably due to the additional imports needed from type annotations, and the refactor. We'll try to improve the situation shortly, thanks for the notice.

**Comment by whyscream** [comment]
HI @pennersr, today we released [3.1.1](https://github.com/mollie/mollie-api-python/releases/tag/3.1.1), which resolves the circular import for `Payment` (and other result classes).
